admin/services/pupils-not-taking-check.service.js

```diff
@@ -0,0 +1,102 @@
+const Pupil = require('../models/pupil')
+
+/** @namespace */
+
+const pupilsNotTakingCheckService = {
+
+  /**
+   * Fetch pupils with reasons to not take the check.
+   * @param schoolId
+   * @returns {Promise.<*>}
+   */
+  fetchPupilsWithReasons: async (schoolId) => {
+    const pupilsWithReasons = await Pupil
+    .find({'attendanceCode': {$exists: true}, 'school': schoolId})
+    .sort('lastName')
+    if (pupilsWithReasons.length > 0) {
+      return pupilsWithReasons
+    }
+  },
+  /**
+   * Sort columns by last name asc/desc.
+   * @param sortField
+   * @param sortDirection
+   * @returns {Promise.<{htmlSortDirection: Array, arrowSortDirection: Array}>}
```

Comment:
I don't see a Promise being returned

Reply:
Indeed, that's wrong. Fixed in next commit.